code/game/atoms.dm

```diff
@@ -433,15 +435,16 @@ var/list/blood_splatter_icons = list()
 	sleep(1)
 	stoplag()
 
-//This is called just before maps and objects are initialized, use it to spawn other mobs/objects
-//effects at world start up without causing runtimes
-/atom/proc/spawn_atom_to_world()
-
 //Called after New if the world is not loaded with TRUE
 //Called from base of New if the world is loaded with FALSE
+//This base must be called or derivatives must set initialized to TRUE to prevent repeat calls
+//Derivatives must not sleep
 /atom/proc/Initialize(mapload)
-	set waitfor = 0

```

Comment:
In regards to removing the waitfor, as useful as it is, it has the potential to cause problems.

Take this call stack for example:
/proc/stoplag
/obj/some/piece/of/shit/Initialize
/obj/some/good/person/Initialize
/datum/subsystem/object/InitializeAtoms

If obj/some/piece/of/shit didn't set initialized to TRUE before sleeping, InitializeAtoms will continue running, possibly resulting in a double Initialization of obj/some/piece/of/shit.